### DISCUSSION TITLE
Home Assistant Blueprints

### DISCUSSION THREAD

**Discussion by madpilot** [amberelectric/public-api#76] [Home Assistant]
Now that Amber is officially in [Home Assistant](https://www.home-assistant.io/integrations/amberelectric/) I thought it would be good to share some automation blueprints!

**Comment by madpilot** [comment]
He's one to get started with. You can use this automation to represent the current price using an RGB bulb.

The colours: 

* Cyan: Extremely low price
* Green: Low price
* Orange: Average price
* Red: High price
* Purple: Price spike

You can change the thresholds when configuring the automation.

[![Open your Home Assistant instance and show the blueprint import dialog with a specific blueprint pre-filled.](https://my.home-assistant.io/badges/blueprint_import.svg)](https://my.home-assistant.io/redirect/blueprint_import/?blueprint_url=https%3A%2F%2Fgithub.com%2Famberelectric%2Fpublic-api%2Fblob%2Fmain%2Fhome-assistant%2Fblueprints%2Flight.yaml)

**Reply by shanelord01** [reply]
This is awesome. Wondering if one could be created for use with https://github.com/alandtse/tesla to enable charging if price drops and disable if over a high threshold? I'm not skilled enough it seems :)

**Reply by purcell-lab** [reply]
@madpilot  Nice concept.

A couple of suggestions for additional features.

As you no doubt are aware Amber has traffic lights defined in its app based on cost and % renewables, it would be great if your integration could expose the upstream Amber recommendations, including the Summary, eg. Traffic light: Red Summary: ‘Use less energy if you can - it’s expensive and dirty now’. (Would be a great text to speech notification for the household :slight_smile: Maybe as attributes of the price sensor. In your blueprint the threshold values are user defined, but aren’t dynamic over time.

![image](https://user-images.githubusercontent.com/79175134/148474455-8c36e07e-082b-4caa-a4d5-00bbd60d811f.png)

Second suggestion or extension really would be to include sets of devices to switch depending on traffic lights:

Green - switch on everything
Yellow - switch off time shift loads (EV Charging, Hot Water Service, dishwasher, dryer, pool pump)
Red - switch off non important loads (turn down Heating / Cooling)
Price Spike - switch off everything except critical loads

Finally, I couldn’t actually get the blueprint to work and received this error:

![image](https://user-images.githubusercontent.com/79175134/148474480-3c826071-83a7-4a3e-bab0-4fc0812ea8ed.png)

**Reply by purcell-lab** [reply]
Sorry, blueprint is working correctly now, just needed to go through an Amber price change on the sensor.

**Reply by madpilot** [reply]
>  As you no doubt are aware Amber has traffic lights defined in its app based on cost and % renewables, it would be great if your integration could expose the upstream Amber recommendations

I ummed and ahhed about this when designing the API, and decided not to do it, because, at the time, the colours were pretty arbitrary, and I assumed most people would just create their own thresholds. However, we are about to roll out a less arbitrary set of colours, that are based on the VMO/DMO prices.

I didn't really want to send the text representations of the status down the wire to keep the payload down - there is a lot of text for only a narrow use-case.

Let me re-think this.

> Second suggestion or extension really would be to include sets of devices to switch depending on traffic lights:

I'm going to make another blueprint for controlling devices, where you just set a threshold. You can use that blueprint to create multiple automation.

> Sorry, blueprint is working correctly now, just needed to go through an Amber price change on the sensor.

Ah, interesting.

**Reply by madpilot** [reply]
This reminds me, I need to update the blueprint, as we are introducing a yellow intermediate colour soon
